pkg/apis/security/v1alpha1/common_validation.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+// Validate validates a StringMatch.
+func (m *StringMatch) Validate(ctx context.Context) *apis.FieldError {
+	notEmpty := 0
+	if m.Exact != "" {
+		notEmpty++
+	}
+	if m.Prefix != "" {
+		notEmpty++
+	}
+	if m.Suffix != "" {
+		notEmpty++
+	}
+	if m.Presence {
+		notEmpty++
+	}
+	if notEmpty > 1 {
```

Comment:
nit: might not be worth the effort, but maybe just saving what was actually set and showing only those as part of the error might be a better UX. Kind of as in here:
https://github.com/knative/eventing/blob/6bfe08d39304ad119310594a7a73f80fbcee676b/pkg/apis/eventing/v1alpha1/trigger_validation.go#L61
no strong opinion though. Only if you think it useful